### PR TITLE
French translation completed

### PR DESCRIPTION
**Pull request by Exorcism0666**
Since I had free time, I have done more than half of the translations to do, I still have a few to complete that I will finish this tonight.

Regarding the Google form, I decided to indicate that it will only be in English because I am not part of the project.

If you see some small mistake, don't hesitate to tell me, I'll try to answer (not like last time).

### CODE REVIEW

**Comment by sdercolin** [comment]
Thanks! Could you check the build errors?
```
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (142, 13): Expecting ','
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (291, 12): Expecting an element
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (291, 13): Expecting ','
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (312, [79](https://github.com/sdercolin/utaformatix3/runs/7454520555?check_suite_focus=true#step:5:80)): Expecting an element
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (321, 132): Expecting '"'
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (405, 36): Expecting '"'
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (451, 104): Expecting '"'
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 77): Expecting an element
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 78): Expecting ','
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 79): Expecting '"'
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (523, 37): Expecting '"'
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (142, 9): Unresolved reference: fr
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (142, 14): Mixing named and positioned arguments is not allowed
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (143, 5): No value passed for parameter 'fr'
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (2[91](https://github.com/sdercolin/utaformatix3/runs/7454520555?check_suite_focus=true#step:5:92), 9): Unresolved reference: fr
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (291, 14): Mixing named and positioned arguments is not allowed
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (2[92](https://github.com/sdercolin/utaformatix3/runs/7454520555?check_suite_focus=true#step:5:93), 5): No value passed for parameter 'fr'
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 13): Unresolved reference: Impossible
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 27): Unresolved reference: convertir
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 41): Unresolved reference: changements
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 56): Unresolved reference: tempo
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (458, 65): Unresolved reference: format
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (540, 43): 'when' expression must be exhaustive, add necessary 'French' branch or 'else' branch instead
e: /home/runner/work/utaformatix3/utaformatix3/src/main/kotlin/ui/strings/Strings.kt: (542, 9): Unresolved reference: Français
```

File: src/main/kotlin/ui/strings/Strings.kt [hunk 1]
```diff
@@ -409,55 +479,67 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
...
         ja = "パラメータをインポートしないことで、より早くインポートする",
         zhCN = "忽略参数，使导入更快",
-        ru = "Игнорировать подробные параметры для ускорения импорта"
+        ru = "Игнорировать подробные параметры для ускорения импорта",
+        fr = "Ignorer les paramètres détaillés pour accélérer l'importation"
     );
 
     fun get(language: Language): String = when (language) {
         English -> en
+        Français -> fr
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:542]: `French`

File: src/main/kotlin/ui/strings/Strings.kt [hunk 2]
```diff
@@ -2,124 +2,144 @@ package ui.strings
...
         zhCN = "该选项可能会增加处理时间。",
-        ru = "Обработка с помощью этого параметра может занять некоторое время."
+        ru = "Обработка с помощью этого параметра может занять некоторое время.",
+        fr = "Le traitement peut prendre un certain temps avec cette option."
     ),
     ProjectZoom(
         en = "Zoom in/out project",
         ja = "プロジェクトをズームイン・ズームアウト",
         zhCN = "缩放工程",
         ru = "Приблизить/отдалить проект"
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:141]: need a comma

File: src/main/kotlin/ui/strings/Strings.kt [hunk 3]
```diff
@@ -177,192 +206,226 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
...
         zhCN = "- 在导入的工程中未找到速度记号。",
-        ru = "- Метки темпа не найдены в импортированном проекте."
+        ru = "- Метки темпа не найдены в импортированном проекте.",
+        fr = "- Aucun tempo n'a été trouvée dans le projet importé."
     ),
     ImportWarningTempoIgnoredInFile(
         en = "- Tempo label ({{bpm}}) at tick {{tick}} in file [{{file}}] was ignored.",
         ja = "- ファイル[{{file}}]の tick {{tick}} にあるテンポ記号（{{bpm}}）を読み込めませんでした。",
         zhCN = "- 未能读取文件[{{file}}]中 tick {{tick}} 处的速度记号（{{bpm}}）。",
         ru = "- Метка темпа ({{bpm}}) на отметке {{tick}} в файле [{{file}}] была проигнорирована."
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:290]: need a comma

File: src/main/kotlin/ui/strings/Strings.kt [hunk 4]
```diff
@@ -177,192 +206,226 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
...
+        ru = "- Метка темпа ({{bpm}}) в предварительных мерах была проигнорирована.",
+        fr = "- Le tempo ({{bpm}}) dans les pré-mesures a été ignorée."
     ),
     ImportWarningTimeSignatureNotFound(
         en = "- No time signature labels found in the imported project.",
         ja = "- 拍子記号が見つかりませんでした。",
         zhCN = "- 在导入的工程中未找到节拍记号。",
-        ru = "- В импортированном проекте не найдено меток временных сигнатур."
+        ru = "- В импортированном проекте не найдено меток временных сигнатур.",
+        fr = "- Aucune signature rythmique n'a trouvé dans le projet importé".
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:312]: The last `".` should be `."`

File: src/main/kotlin/ui/strings/Strings.kt [hunk 5]
```diff
@@ -177,192 +206,226 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
...
+        ru = "Проект для VOCALOID2",
+        fr = "Projet pour VOCALOID2"
     ),
     VocaloidMidiFormatDescription(
         en = "Project for VOCALOID1",
         ja = "VOCALOID1 プロジェクト",
         zhCN = "VOCALOID1 工程",
-        ru = "Проект для VOCALOID1"
+        ru = "Проект для VOCALOID1",
+        fr = "Projet pour VOCALOID1
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:405]: Missing `"`

File: src/main/kotlin/ui/strings/Strings.kt [hunk 6]
```diff
@@ -381,25 +446,30 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
         zhCN = "所有音符的音素都被设为了\"a\"。请使用VOCALOID5菜单中的「任务」->「发音符号匹配」功能来重置音素。",
         ru = "Фонемы всех нот были установлены на \"a\"." +
             " Пожалуйста, воспользуйтесь \"Job\" -> \"Convert Phonemes to Match Languages\"" +
-            " в меню VOCALOID5 чтобы сбросить их."
+            " в меню VOCALOID5 чтобы сбросить их.",
+        fr = "Les phonèmes de toutes les notes ont été réglés sur \"a\". Veuillez utiliser \"Lyrics\" ->" +
+            " \"Convert Phonemes to Match Languages\" dans le menu de VOCALOID5 pour les réinitialiser.
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:451]: Missing `"` at the end

File: src/main/kotlin/ui/strings/Strings.kt [hunk 7]
```diff
@@ -381,25 +446,30 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
...
     ),
     ExportNotificationTempoChangeIgnored(
         en = "Could not convert tempo changes to the target format.",
         ja = "テンポの変更を出力することができませんでした。",
         zhCN = "未能将速度的变化导出到目标格式。",
-        ru = "Невозможно конвертировать изменения темпа в целевой формат."
+        ru = "Невозможно конвертировать изменения темпа в целевой формат.",
+        fr =Impossible de convertir les changements de tempo au format cible."
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:458]: Missing `"` at the beginning

File: src/main/kotlin/ui/strings/Strings.kt [hunk 8]
```diff
@@ -409,55 +479,67 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
...
+        ru = "Максимальная длина, подлежащая обработке (исключительная)",
+        fr = "Longueur maximale à traiter (exclusive)"
     ),
     SlightRestsFillingThresholdItem(
         en = "1/{{denominator}} note",
         ja = "{{denominator}}分音符",
         zhCN = "{{denominator}}分音符",
-        ru = "1/{{denominator}} ноты"
+        ru = "1/{{denominator}} ноты",
+        fr = "1/{{denominator}} note
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:523]: Missing `"`

File: src/main/kotlin/ui/strings/Strings.kt [hunk 9]
```diff
@@ -2,124 +2,144 @@ package ui.strings
 
 import io.MusicXml.MUSIC_XML_VERSION
 import ui.strings.Language.English
+import ui.strings.Language.French
 import ui.strings.Language.Japanese
 import ui.strings.Language.Russian
 import ui.strings.Language.SimplifiedChinese
 
-enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: String) {
+enum class Strings(val en: String, val fr: String, val ja: String, val zhCN: String, val ru: String) {
```
Comment (sdercolin) [inline src/main/kotlin/ui/strings/Strings.kt:10]: Could you move `fr` to the last one, so that the order is the same as the following assignments

**Comment by sdercolin** [comment]
```
Q: J'utilise la conversion de hauteur d'UtaFormatix, mais cela sonne mal après la conversion.
R: Les mêmes données de hauteur sonnent différemment dans différents éditeurs en raison de nombreux facteurs, y compris la transition de hauteur (hauteur
connexion entre deux notes), vibrato par défaut, vitesse de note par défaut (longueur des consonnes) etc. Spécialement, si les données de hauteur converties contiennent une transition de hauteur (par exemple ust), vous pouvez essayer d'ajuster les paramètres de transition de hauteur (ou utiliser
mode pitch-snap/robot) dans l'éditeur cible pour éviter la superposition.
```
Seems like there are some unexpected spaces. 
```hauteur
connexion entre deux notes```
```
ou utiliser
mode pitch-snap/robot
```

Please fix it in your gist.

File: src/main/kotlin/ui/strings/Strings.kt [hunk 10]
```diff
@@ -177,192 +206,226 @@ enum class Strings(val en: String, val ja: String, val zhCN: String, val ru: Str
...
     ),
     ImportWarningTimeSignatureIgnoredInTrack(
         en = "- Time signature label ({{timeSignature}}) at measure {{measure}}" +
             " in Track {{number}}: [{{name}}] was ignored.",
         ja = "- トラック{{number}}：[{{name}}]の小節{{measure}}にある拍子記号（{{timeSignature}}）を読み込めませんでした。",
         zhCN = "- 未能读取音轨{{number}}: [{{name}}]中小节{{measure}}处的节拍记号（{{timeSignature}}）。",
         ru = "- Метка временной сигнатуры ({{TimeSignature}}) при измерении {{measure}}" +
-            " в треке {{number}}: [{{name}}] была проигнорирована."
+            " в треке {{number}}: [{{name}}] была проигнорирована.",
+        fr = "La signature rythmique ({{timeSignature}}) à la mesure {{measure}} de la piste {{number}} : [{{name}}]] a été ignoré.
```
Comment (colin-weng) [inline src/main/kotlin/ui/strings/Strings.kt:321]: One more missing `"` here!

Comment (colin-weng) [reply src/main/kotlin/ui/strings/Strings.kt]: It's ok~
You might find an IDE useful..

**Review by sdercolin — APPROVED** [review]
LGTM. merged to develop branch. You can check the result at beta app: https://utaformatix-beta.herokuapp.com/

**Pull request merged**